activemq-broker/src/main/java/org/apache/activemq/plugin/SubQueueSelectorCacheBroker.java

```diff
@@ -369,7 +369,10 @@ public SubSelectorClassObjectInputStream(InputStream is) throws IOException {
 
         @Override
         protected Class<?> resolveClass(ObjectStreamClass desc) throws IOException, ClassNotFoundException {
-            if (!(desc.getName().equals("java.lang.String") || desc.getName().startsWith("java.util."))) {
+            if (!(desc.getName().startsWith("java.lang.")
+                    || desc.getName().startsWith("com.thoughtworks.xstream")
```

Comment:
Is there a way to check if xstream is enabled, and if so to include this check, otherwise leave it out?

Reply:
It could be enabled when using HTTP transport connector.

Reply:
I think it's worth to keep it (just in case for http transport connector). Thoughts ?